src/integrations/fhir/resolvers.ts

```diff
@@ -140,4 +150,158 @@ export const phoneContactResolver = (telecoms: IContactPoint[] | undefined): str
             onlyPhonesNotOld[0]
 
     return chosenPhone?.value
-}
\ No newline at end of file
+}
+
+/**
+ * Extract the resource list from a bundle.
+ * XXX This might not be needed anymore, when using the "flat: true" request option.
+ * @param bundle
+ */
+export const bundleResourceList = (bundle: IBundle): IResourceList[] =>
+    bundle.entry !== undefined ?
+        bundle.entry.flatMap(e => e.resource !== undefined ? [e.resource] : []) :
+        []
+
+/**
+ * Validate that response is a resource list array.
+ * Use this when processing requests made with option "flat: true", before validating/extracting the actual desired resource from returned list
+ * @param resp
+ */
+export const iResourceListFromArray = (resp: unknown[]): IResourceList[] => resp.map(r => validateOrThrow(R4.RTTI_ResourceList.decode(r)))
+
+/**
+ * Return first resource of type IPractitionerRole from given resourcelisting.
+ * @param resourcelisting
+ */
+export const iPractitionerRoleFromListing = (resourcelisting: IResourceList[]): IPractitionerRole | undefined => {
+    for(const resource of resourcelisting) {
+        if(R4.RTTI_PractitionerRole.is(resource)) {
+            return resource
+        }
+    }
+    return undefined
+}
+
+export const iPractitionerFromListing = (resourcelisting: IResourceList[]): IPractitioner | undefined => {
+    for(const resource of resourcelisting) {
+        if(R4.RTTI_Practitioner.is(resource)) {
+            return resource
+        }
+    }
+    return undefined
+}
+
+/**
+ * Return first identifier of type "hpr nummer" from given list of identifiers.
+ * "hpr number" is an official norwegian healthcare persons id-number. This has the OID "urn:oid:2.16.578.1.12.4.1.4.4".
+ * https://www.ehelse.no/teknisk-dokumentasjon/oid-identifikatorserier-i-helse-og-omsorgstjenesten#Nasjonale%20identifikatorserier%20for%20personer
+ */
+export const hprNumberFromIdentifiers = (identifiers: IIdentifier[]): HprNumber | undefined => {
+    for(const identifier of identifiers) {
+        if(
+            identifier.system === "urn:oid:2.16.578.1.12.4.1.4.4" &&
```

Comment:
Fint om du oppretter en konstant for denne IOD-en på samme måte som for fnrOid, dnrOid, og orgnrOid :)

Reply:
Einig, har gjort det no :)